robot/session_manager.go

```diff
@@ -104,6 +106,16 @@ func (m *SessionManager) expireLoop(ctx context.Context) {
 					}()
 					res, err := m.robot.ResourceByName(resName)
 					if err != nil {
+						// It's possible at this point that the robot is Closing, the
+						// resource manager has already been closed, and the resource
+						// associated with the session has been removed from the graph and
+						// cannot be found. If the error is a not found error and the
+						// context has errored, return without appending to resourceErrs
+						// and set serverClosing to true.
+						if strings.Contains(err.Error(), resource.NewNotFoundError(resName).Error()) && ctx.Err() != nil {
```

Comment:
I can't use `errors.Is` here because `NewNotFoundError` does not return an error that implements `Is`.